cfme/fixtures/parallelizer/__init__.py

```diff
@@ -496,8 +517,30 @@ def pytest_sessionfinish(self):
         self.zmq_ctx.destroy()
 
     def _test_item_generator(self):
-        for tests in self._modscope_item_generator():
-            yield tests
+        if self.serial_collection and not self.collection:
+            for tests in self._serial_item_generator():
+                yield tests
+        elif self.collection and not self.serial_collection:
+            for tests in self._modscope_item_generator():
+                yield tests
+        else:
+            for tests in chain(self._serial_item_generator(), self._modscope_item_generator()):
```

Comment:
what is above `if` statement for ?

Reply:
This is admittedly a hacky way to do this, but I couldn't come up with a better way. The issue was with all the generators that the parallelizer uses. The `serial` collection is kept separate from the normal collection, so we don't want to call e.g. the serial generator if the pytest collection contains no serially marked tests (and vice versa). If it contains both, then we must chain the generators together.

Looking at this again, I'm wondering whether or not we need the first two if statements, maybe we could get away with just the `chain`?

Reply:
Testing it out now, the issue was that without the `if/elif`, if you collect all serial (or all normal) the tests aren't properly collected and you end up with an empty test suite.